packages/flutter/lib/src/widgets/image.dart

```diff
@@ -108,6 +108,80 @@ Future<void> precacheImage(
   return completer.future;
 }
 
+/// Signature used by [Image.frameBuilder] to control the widget that will be
+/// used when an [Image] is built.
+///
+/// The `child` argument contains the default widget that would be rendered if
+/// no [Image.frameBuilder] had been specified. In other words, returning
+/// `child` directly from this method will yield the same result as this method
+/// having not been consulted. This argument is guaranteed to be non-null.
+/// Typically, this builder will wrap the `child` widget in some way and return
+/// the wrapped widget.
+///
+/// The `frame` argument specifies the index of the current image frame being
+/// rendered. It will be null before the first image frame is ready, and zero
+/// for the first image frame. For single-frame images, it will never be greater
+/// than zero. For multi-frame images (such as animated GIFs), it will increase
+/// by one every time a new image frame is shown (including when the image
+/// animates in a loop).
+///
+/// The `wasSynchronouslyLoaded` argument specifies whether the image was
+/// available synchronously (immediately). If this is true, then the image was
+/// able to be painted immediately. If this is false, then there was one or more
+/// frames where the image wasn't yet available to be painted. For multi-frame
+/// images, the value of this argument is dictated the first frame. In other
```

Comment:
"is dictated the first frame" doesn't make sense.

Also, this paragraph uses the word "frame" for two different purposes without clarification, which is confusing.

Reply:
Done